rising/transforms/compose.py

```diff
@@ -227,3 +234,54 @@ def forward(self, *seq_like, **map_like) -> Union[Sequence, Mapping]:
             if rand[idx] > self.dropout[idx]:
                 data = self.transform_call(data, self.transforms[idx])
         return data
+
+
+class OneOf(AbstractTransform):
+    """
+    Apply one of the given transforms.
+    """
+
+    def __init__(self, *transforms: Union[AbstractTransform, Sequence[AbstractTransform]],
+                 weights: Optional[Sequence[float]] = None,
+                 p: float = 1.,
+                 transform_call: Callable[[Any, Callable], Any] = dict_call):
+        """
+        Args:
+            *transforms: transforms to choose from
+            weights: additional weights for transforms
+            p: probability that one transform i applied
+            transform_call: function which determines how transforms are
+                called. By default Mappings and Sequences are unpacked
+                during the transform.
+
+        References:
+            (https://github.com/fepegar/torchio/blob/687aa8d73e9377f4e6f7db7996
+            09083c2d092eae/torchio/transforms/augmentation/composition.py)
+        """
+        super().__init__(grad=True)
+        if len(transforms) > 0 and isinstance(transforms[0], Sequence):
+            transforms = transforms[0]
+        if not transforms:
+            raise ValueError("At least one transformation needs to be selected.")
+        self.transforms = transforms
+
+        if weights is not None and len(weights) != len(transforms):
+            raise ValueError("If weights are porvided, every transform needs a weight. "
+                             f"Found {len(weights)} weights and {len(transforms)} transforms")
+        if weights is None:
+            self.weights = torch.tensor(
+                [1 / len(self.transforms)] * len(self.transforms))
+        else:
+            if not isinstance(weights, torch.Tensor):
+                self.weights = torch.tensor(weights)
+            else:
+                self.weights = weights
```

Comment:
can we transform the weights parameter and assign it all in one line after checking the length of weights is the same as the number of transforms?